node/chainSimulator/chainSimulator.go

```diff
@@ -40,25 +48,34 @@ type transactionWithResult struct {
 
 // ArgsChainSimulator holds the arguments needed to create a new instance of simulator
 type ArgsChainSimulator struct {
-	BypassTxSignatureCheck   bool
-	TempDir                  string
-	PathToInitialConfig      string
-	NumOfShards              uint32
-	MinNodesPerShard         uint32
-	MetaChainMinNodes        uint32
-	NumNodesWaitingListShard uint32
-	NumNodesWaitingListMeta  uint32
-	GenesisTimestamp         int64
-	InitialRound             int64
-	InitialEpoch             uint32
-	InitialNonce             uint64
-	RoundDurationInMillis    uint64
-	RoundsPerEpoch           core.OptionalUint64
-	ApiInterface             components.APIConfigurator
-	AlterConfigsFunction     func(cfg *config.Configs)
-}
-
-type simulator struct {
+	BypassTxSignatureCheck      bool
+	TempDir                     string
+	PathToInitialConfig         string
+	NumOfShards                 uint32
+	MinNodesPerShard            uint32
+	ConsensusGroupSize          uint32
+	MetaChainMinNodes           uint32
+	MetaChainConsensusGroupSize uint32
+	NumNodesWaitingListShard    uint32
+	NumNodesWaitingListMeta     uint32
+	GenesisTimestamp            int64
+	InitialRound                int64
+	InitialEpoch                uint32
+	InitialNonce                uint64
+	RoundDurationInMillis       uint64
+	RoundsPerEpoch              core.OptionalUint64
+	ApiInterface                components.APIConfigurator
+	AlterConfigsFunction        func(cfg *config.Configs)
+	SimulatorType               string
+	CreateGenesisNodesSetup     func(nodesFilePath string, addressPubkeyConverter core.PubkeyConverter, validatorPubkeyConverter core.PubkeyConverter, genesisMaxNumShards uint32) (mxChainSharding.GenesisNodesSetupHandler, error)
+	CreateRatingsData           func(arg rating.RatingsDataArg) (processing.RatingsInfoHandler, error)
+	CreateIncomingHeaderHandler func(config *config.NotifierConfig, dataPool dataRetriever.PoolsHolder, mainChainNotarizationStartRound uint64, runTypeComponents factory.RunTypeComponentsHolder) (processing.IncomingHeaderSubscriber, error)
+	GetRunTypeComponents        func(args runType.ArgsRunTypeComponents) (factory.RunTypeComponentsHolder, error)
+	NodeFactory                 node.NodeFactory
+	ChainProcessorFactory       ChainProcessorFactory
+}
+
+type Simulator struct {
```

Comment:
Why did you export this?
This should not be exported

Reply:
You don't need it. You just exported a struct with full private members. That struct should only be created through its constructor. Please revert back

Reply:
reverted